From adf4cf4cb8c0e8c8753240d63aa6d68d70b8f20f Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@dereferenced.net>
Date: Mon, 7 Nov 2016 21:21:13 -0700
Subject: [PATCH] Forcefully disable getentropy and clock_gettime on macOS due
 to a SDK bug

getentropy and clock_gettime were added in macOS 10.12, but the
__OSX_AVAILABLE_STARTING macro was left off of the declaration in the SDK
header, so the -mmacosx-version-min flag is ignored. This results in a tor
that links to symbols only available on 10.12 and later.

A cleaner, possibly upstreamable fix would be to check whether
(MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_12) and skip these
function tests.
---
 configure.ac | 2 --
 1 file changed, 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 37827d6..e1b146a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -394,14 +394,12 @@ AC_CHECK_FUNCS(
         accept4 \
         backtrace \
         backtrace_symbols_fd \
-        clock_gettime \
 	eventfd \
 	explicit_bzero \
 	timingsafe_memcmp \
         flock \
         ftime \
         getaddrinfo \
-        getentropy \
         getifaddrs \
         getpass \
         getrlimit \
-- 
2.10.0

